### PR TITLE
tests.cc-wrapper: add atomics test

### PR DESCRIPTION
**Pull request by Artturin**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/test/cc-wrapper/default.nix [hunk 1]
```diff
@@ -37,6 +38,14 @@ in stdenv.mkDerivation {
     $CXX -o include-cxxabi ${./include-cxxabi.cc}
     ${emulator} ./include-cxxabi
 
+    # cxx doesn't have libatomics.so
```
Comment (Mindavi) [inline pkgs/test/cc-wrapper/default.nix:41]:
```suggestion
    # cxx doesn't have libatomic.so
```

**Review by Mindavi — APPROVED** [review]
Diff LGTM

**Comment by Artturin** [comment]
@ofborg build tests.cc-wrapper.supported

**Comment by Artturin** [comment]
aarch64-linux

```
error: builder for '/nix/store/r7n6j5m0r655jlhyn21kcchy93dawbvw-cc-wrapper-test-gcc-7.5.0.drv' failed with exit code 1;
       last 10 log lines:
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: (.text+0x84c): undefined reference to `__aarch64_ldadd4_relax'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: (.text+0x898): undefined reference to `__aarch64_ldadd4_relax'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: (.text+0x8b0): undefined reference to `__aarch64_ldadd4_relax'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: /nix/store/8qcc8pnxgmskwlsbjrjn3z37sws4g0hd-glibc-2.38-23-static/lib/libc.a(dl-runtime.o): in function `_dl_fixup':
       > (.text+0x164): undefined reference to `__aarch64_swp4_rel'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: /nix/store/8qcc8pnxgmskwlsbjrjn3z37sws4g0hd-glibc-2.38-23-static/lib/libc.a(dl-runtime.o): in function `_dl_profile_fixup':
       > (.text+0x410): undefined reference to `__aarch64_swp4_rel'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: /nix/store/8qcc8pnxgmskwlsbjrjn3z37sws4g0hd-glibc-2.38-23-static/lib/libc.a(dl-sym.o): in function `do_sym':
       > (.text+0x244): undefined reference to `__aarch64_swp4_rel'
       > collect2: error: ld returned 1 exit status
       For full logs, run 'nix log /nix/store/r7n6j5m0r655jlhyn21kcchy93dawbvw-cc-wrapper-test-gcc-7.5.0.drv'.
```

```
error: builder for '/nix/store/s47sdkz4h7h0ki0jqb32qmmddjxqpiji-cc-wrapper-test-gcc-6.5.0.drv' failed with exit code 1;
       last 10 log lines:
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: (.text+0x84c): undefined reference to `__aarch64_ldadd4_relax'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: (.text+0x898): undefined reference to `__aarch64_ldadd4_relax'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: (.text+0x8b0): undefined reference to `__aarch64_ldadd4_relax'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: /nix/store/8qcc8pnxgmskwlsbjrjn3z37sws4g0hd-glibc-2.38-23-static/lib/libc.a(dl-runtime.o): in function `_dl_fixup':
       > (.text+0x164): undefined reference to `__aarch64_swp4_rel'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: /nix/store/8qcc8pnxgmskwlsbjrjn3z37sws4g0hd-glibc-2.38-23-static/lib/libc.a(dl-runtime.o): in function `_dl_profile_fixup':
       > (.text+0x410): undefined reference to `__aarch64_swp4_rel'
       > /nix/store/q5ivgn27vl58cahzng642baihd9sbssp-binutils-2.40/bin/ld: /nix/store/8qcc8pnxgmskwlsbjrjn3z37sws4g0hd-glibc-2.38-23-static/lib/libc.a(dl-sym.o): in function `do_sym':
       > (.text+0x244): undefined reference to `__aarch64_swp4_rel'
       > collect2: error: ld returned 1 exit status
       For full logs, run 'nix log /nix/store/s47sdkz4h7h0ki0jqb32qmmddjxqpiji-cc-wrapper-test-gcc-6.5.0.drv'.
```

**Pull request merged**